apps/meteor/app/lib/server/functions/insertMessage.ts

```diff
@@ -1,9 +1,15 @@
 import { Messages, Rooms } from '@rocket.chat/models';
+import type { IMessage, IRoom, IUser } from '@rocket.chat/core-typings';
 
 import { validateMessage, prepareMessageObject } from './sendMessage';
 import { parseUrlsInMessage } from './parseUrlsInMessage';
 
-export const insertMessage = async function (user, message, rid, upsert = false) {
+export const insertMessage = async function (
+	user: IUser,
```

Comment:
`question` Must it be a full `IUser`? Making it like `Pick<IUser, '_id' | 'username'>` would avoid the assertion at https://github.com/RocketChat/Rocket.Chat/pull/28835/files#diff-4c5584bef947bc711ab14cc203bb3694c22c5a939ca4d4eed9340c9c45750b93R625

Reply:
🤔 let me try